### PR TITLE
Cart Fingerprint

### PR DESCRIPTION
**Pull request by glennjacobs**
This idea introduces a cart fingerprint so you can quickly identify if a cart has changed.

- [x] Tests
- [x] Docs
- [x] Changelog

### CODE REVIEW

File: docs/core/upgrading.md [hunk 1]
```diff
@@ -22,7 +22,17 @@ php artisan lunar:hub:install
 
 ## Support Policy
 
-Lunar currently provides bug fixes and security updates for only the latest minor release, e.g. `0.3`. 
+Lunar currently provides bug fixes and security updates for only the latest minor release, e.g. `0.3`.
+
+## [Unreleased]
+
+### High impact
```
Comment (glennjacobs) [inline docs/core/upgrading.md:29]: Is it really a high impact change? Because we set a default, so it won't break if it's omitted.

**Review by alecritson — APPROVED** [review]

**Comment by wychoong** [comment]
a side affect of this PR where the casting of `meta` is now `AsArrayObject` instead of `'object'` makes accessing meta attributes more expectable, without the need of `(array)` cast or `is_object/is_array` check.

@glennjacobs would you consider to make all model's meta attribute `AsArrayObject`?

**Comment by glennjacobs** [comment]
Yeah, I actually wanted to change those. I think there's other models I wanted to change too. I'll review more when on the laptop.

**Comment by glennjacobs** [comment]
> a side affect of this PR where the casting of `meta` is now `AsArrayObject` instead of `'object'` makes accessing meta attributes more expectable, without the need of `(array)` cast or `is_object/is_array` check.
> 
> @glennjacobs would you consider to make all model's meta attribute `AsArrayObject`?

Did you want to do a PR for this?

**Comment by wychoong** [comment]
> > a side affect of this PR where the casting of `meta` is now `AsArrayObject` instead of `'object'` makes accessing meta attributes more expectable, without the need of `(array)` cast or `is_object/is_array` check.
> > @glennjacobs would you consider to make all model's meta attribute `AsArrayObject`?
> 
> Did you want to do a PR for this?

I can create a PR with fixes for features covered by test only, if thats ok

**Comment by glennjacobs** [comment]
> > > a side affect of this PR where the casting of `meta` is now `AsArrayObject` instead of `'object'` makes accessing meta attributes more expectable, without the need of `(array)` cast or `is_object/is_array` check.
> > > @glennjacobs would you consider to make all model's meta attribute `AsArrayObject`?
> > 
> > 
> > Did you want to do a PR for this?
> 
> I can create a PR with fixes for features covered by test only, if thats ok

Got for it. I'm happy to contribute if I think further test coverage is needed.

**Pull request merged**